### ISSUE TITLE
feature request: option to track the validators

### ISSUE THREAD

**Issue by llllvvuu** [a16z/helios#180]
1) From the `--checkpoint` we have `BeaconBlock.state_root`. Get the full `BeaconState` from `/eth/v2/debug/beacon/states/{state_id}`, or if not available, from the `/eth/v1/beacon/states/{state_id}` endpoints. Verify against the `state_root`.
2) Implement [state_transition(state: BeaconState, signed_block: SignedBeaconBlock, validate_result: bool=True) -> None](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#beacon-chain-state-transition-function).
3) In [`consensus.rs:is_finality_proof_valid()`](https://github.com/a16z/helios/blob/master/consensus/src/consensus.rs#L562), implement [`process_justification_and_finalization(state: BeaconState) -> None`](https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#justification-and-finalization)

**Comment by ncitron** [comment]
Yeah this is a good idea. I do think we can make it a bit more like the cosmos LC protocol though for efficiency reasons. We don't need to track the validator set fully but can rather fetch it once from the checkpoint and then check future headers with that set. Every once in a while we need to refresh the validator set to ensure that that many validators haven't churned, but you only need to do this infrequently. This means we don't need to do the state transition every slot, but as long as we are smart about setting the set refresh cadence, it should be just as secure.

**Comment by mahmudsudo** [comment]
Hi , can I take on this ?

**Comment by ncitron** [comment]
I'm not sure we're interested in pursuing this at the moment. I think verifying all of consensus would be much to heavyweight in practice, but if you think its possible and have some data on how much bandwidth and compute we'd need I could be convinced.

**Comment by mahmudsudo** [comment]
ok then , i would pick a different one
